src/GameLogic/Store/CampaignBrowserStore.ts

```diff
@@ -0,0 +1,54 @@
+import * as localforage from 'localforage';
+import {CampaignStore} from './CampaignStore';
+import {Campaign} from '../Campaign';
+import {CampaignMetadata} from './CampaignMetadata';
+import {PakoUtils} from '../../Utils/PakoUtils';
+import {CampaignSerializer} from '../Serialization/CampaignSerializer';
+import {CampaignDeserializer} from '../Serialization/CampaignDeserializer';
+
+type StoreRow = [string, string, string];
+
+export class CampaignBrowserStore implements CampaignStore {
```

Comment:
Again, this might just be me looking too far into the future, but I presume that the game will start with a campaign, they won't just all be in local storage? I assume this is for supporting custom campaigns. Or will it be like DROD where there's a core data file provided that starts with the game's campaign already stored in it?

Maybe I'm confusing localforage for being something using the browser's local storage, when it's not quite like that.

Reply:
Localforage is a library that gives you a single API to access any storage in the browser - localstorage, indexeddb or websql, depending on what's available. I went with it rather than using  just localstorage is because of the size limitations. Localstorage can only hold a few MB of data, while indexeddb is essentially limitless.

As for the storage, pretty much yes - the campaign will be in the game somewhere, somehow, and the store is primarily useful for custom campaigns. Whether the main campaign will also be loaded into the store or treated differently - that's probably a question that will be answered when we'll be thinking about the UI.